sklearn/metrics/tests/test_common.py

```diff
@@ -148,6 +147,14 @@
     "cohen_kappa_score": cohen_kappa_score,
 }
 
+# Metrics are tested with numpy's assert_array. Therefor box-shaped outputs
+# are required.
+CURVE_METRICS = {
+    "roc_curve": lambda *args, **kwargs: roc_curve(*args, **kwargs)[:2],
```

Comment:
you're throwing away the thresholds, right? Which makes sense, but I think the comment should say that. That's also not only about the shape right? The thresholds probably don't behave like a metric.

Reply:
Yes, I am throwing away the thresholds here. But no, I am only really doing it because of shape, as `thresholds` will have a different dimension compared to other returned values.
  
If I recall correctly the `thresholds` on their own should pass all the tests. I will test if we can keep the `thresholds` by padding the result into a box-shaped array rather than throwing stuff away.